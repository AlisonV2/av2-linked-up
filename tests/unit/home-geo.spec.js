import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import HomeGeoloc from '@/components/home/Geoloc';
import AppTitle from '@/components/app/Title';
import AppButton from '@/components/app/Button';

const actions = {
  getArtistsByCity: jest.fn(),
};

const store = createStore({
  actions,
});

const $router = {
  push: jest.fn(),
};

const config = {
  global: {
    plugins: [store],
    stubs: {
      RouterLink: RouterLinkStub,
      'app-title': AppTitle,
      'app-button': AppButton,
    },
    mocks: {
      $router,
    },
  },
};

/**
 * @module HomeGeolocTest
 */
describe('HomeGeoloc.vue', () => {
  it('HomeGeoloc Snapshot', () => {
    const component = shallowMount(HomeGeoloc, config);
    expect(component.element).toMatchSnapshot();
  });

  it('handleSubmit should be called', async () => {
    const component = shallowMount(HomeGeoloc, config);
    component.find('form').trigger('submit.prevent');
    await component.vm.$nextTick();
    expect(component.emitted().handleSubmit).toHaveBeenCalled;
  });
});
