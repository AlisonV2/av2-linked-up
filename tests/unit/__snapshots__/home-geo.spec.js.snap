// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeGeoloc.vue HomeGeoloc Snapshot 1`] = `
<form
  class="col-12 col-md-6 geoloc"
>
  <div
    class="col-12 text-center mb-4 app-title"
  >
    <h2
      class="title"
    >
      
      Looking for an artist? 
      
    </h2>
  </div>
  <div
    class="input-group mb-3"
  >
    <input
      autocomplete=""
      class="form-control"
      id="autocomplete"
      placeholder="Enter your city"
      type="text"
    />
    <span
      class="input-group-text material-icons"
    >
      place
    </span>
  </div>
  <div
    class="btn-center"
  >
    <button
      class="btn"
    >
      
      Search
      
    </button>
  </div>
  <a>
    <div
      class="col-12 text-center mb-4 app-title"
    >
      <h2
        class="title"
      >
        
        Or search by styles 
        <i
          class="bi bi-arrow-right"
        />
        
      </h2>
    </div>
  </a>
</form>
`;
